pom.xml

```diff
@@ -110,7 +110,7 @@
     <antlr4.version>4.9.3</antlr4.version>
     <apache-directory-server.version>1.5.7</apache-directory-server.version>
     <!-- Include arrow for LlapOutputFormatService -->
-    <arrow.version>12.0.0</arrow.version>
+    <arrow.version>16.1.0</arrow.version>
```

Comment:
I guess this Apache Arrow is compiled by a different JDK version

```
[2024-12-29T18:31:23.075Z] Caused by: java.lang.NoSuchMethodError: java.nio.ByteBuffer.position(I)Ljava/nio/ByteBuffer;
[2024-12-29T18:31:23.075Z] 	at org.apache.arrow.memory.ArrowBuf.setBytes(ArrowBuf.java:850)
[2024-12-29T18:31:23.075Z] 	at io.netty.buffer.NettyArrowBuf.setBytes(NettyArrowBuf.java:278)
[2024-12-29T18:31:23.075Z] 	at io.netty.buffer.MutableWrappedByteBuf.setBytes(MutableWrappedByteBuf.java:341)
[2024-12-29T18:31:23.075Z] 	at io.netty.buffer.ExpandableByteBuf.setBytes(ExpandableByteBuf.java:27)
[2024-12-29T18:31:23.075Z] 	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1113)
```

Reply:
@okumin yes, you are right. I was looking into the same and it seems arrow v16.0.0+ is compiled with jdk9. 
Have changed the arrow target version to 15.0.0 in [cd3eb8f](https://github.com/apache/hive/pull/5592/commits/cd3eb8fb1d832c68d908d33357ccf642e4a82980) which seems compatible.

Reply:
my bad, arrow v15.0.0 was producing some license-related issues causing compilation failure.
Instead, have upgraded to v16.0.0